### PR TITLE
refactor(build): removed component scans from configurations

### LINKED ISSUE #1399: @ComponentScan on org.activiti in starters is too broad
The audit service in particular has a dependency on the engine in the effective pom. Is it needed? Can it be excluded?

If you look at the JpaApplication in the tests in the starter for audit it does a broad componentscan on org.activiti. If you restrict it just to the audit package then it doesn't work. Need to debug and figure out what packages needed.

### PR DESCRIPTION
**Pull request by igdianov**
This PR fixes Spring Boot bean override problems by removing component scans from configuration classes and adds missing @ConditionalOnMissingBean annotation on beans created in Activiti Runtime Spring Boot Auto-configurations

Depends on https://github.com/Activiti/activiti-cloud-service-common/pull/207

Fixes https://github.com/Activiti/Activiti/issues/1399

### CODE REVIEW

File: activiti-cloud-services-org/activiti-cloud-services-org-rest/pom.xml [hunk 1]
```diff
@@ -51,6 +51,10 @@
       <groupId>com.fasterxml.jackson.core</groupId>
       <artifactId>jackson-annotations</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-validation</artifactId>
```
Comment (erdemedeiros) [inline activiti-cloud-services-org/activiti-cloud-services-org-rest/pom.xml]: @igdianov, Is it ok to have this starter here?

**Review by erdemedeiros — CHANGES_REQUESTED** [review]
@igdianov, just one question here: as we are moving to have dependencies to other starters only in our starter, is it fine to have `spring-boot-starter-validation` at this level?

File: activiti-cloud-services-org/activiti-cloud-services-org-rest/pom.xml (hunk 1 above)
Comment (igdianov) [reply activiti-cloud-services-org/activiti-cloud-services-org-rest/pom.xml]: I think it is OK, but I will double check

Comment (igdianov) [reply activiti-cloud-services-org/activiti-cloud-services-org-rest/pom.xml]: @erdemedeiros I have removed `spring-boot-starter-validation` from compile scope

**Review by erdemedeiros — APPROVED** [review]

**Pull request merged**